.github/workflows/linux.yaml

```diff
@@ -0,0 +1,103 @@
+name: Linux
+
+on: [push, pull_request]
+
+env:
+  REGISTRY_HOST: pcr-internal.puppet.net
+  DOCKER_REGISTRY: pe-and-platform
+  BOLT_SUDO_USER: true
+
+jobs:
+
+  modules:
+    name: Modules
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v1
+      - name: Setup Ruby
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: '2.5.x'
+      - name: Install bundler
+        run: |
+          gem install bundler
+          bundle config path vendor/bundle
+      - name: Cache gems
+        id: cache
+        uses: actions/cache@v1
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile') }}
+      - name: Install gems
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: bundle install --jobs 4 --retry 3
+      - name: Run module tests
+        run: bundle exec rake github:modules
+
+  unit:
+    name: Unit
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v1
+      - name: Setup Ruby
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: '2.5.x'
+      - name: Install bundler
+        run: |
+          gem install bundler
+          bundle config path vendor/bundle
+      - name: Cache gems
+        id: cache
+        uses: actions/cache@v1
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile') }}
+      - name: Install gems
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: bundle install --jobs 4 --retry 3
+      - name: Pre-test setup
+        run: |
+          sudo curl -L https://github.com/docker/compose/releases/download/1.23.0/docker-compose-`uname -s`-`uname -m` -o /usr/local/bin/docker-compose
+          sudo chmod +x /usr/local/bin/docker-compose
+          echo 'runner:runner' | sudo chpasswd
+          sudo sh -c "echo 'Defaults authenticate' >> /etc/sudoers"
```

Comment:
I think we're fine to leave these, but I'm a little curious if they're necessary in the action containers. Basically the `travis` user in TravisCI automatically doesn't need authentication to use `sudo`, and for some tests we *need* the user we're escalating as to ask for a password. If 'runner' already asks for a password when using sudo, and if the user already has a predictable password, then we can get rid of some of these steps.

Reply:
That applies to this line and the next one.

Reply:
The `runner` user in GitHub Actions uses passwordless sudo just like Travis, so we'd need to keep these steps in.